include/materials/NEMLStress.h

```diff
@@ -17,62 +17,49 @@
-   *input file
-   * @param nemlNames variables names from input file
-   * @return vector containing the values that will be substituted in for the each nemlNames
-   **/
-  std::vector<Real> constructNemlSubstitutionList(const std::vector<std::string> & nemlNames) const;
+protected:
+  /// list of {variable} names for substitution in the XML file
+  std::vector<std::string> _neml_variable_iname;
+
+  /// Vector of values to substitute in
+  std::vector<Real> _neml_variable_value;
+
+  /// Number of {variables{ to substitute in the XML}}
+  const std::size_t _nvars;
+
+  /// set of variables in the XML file
+  std::set<std::string> _xml_vars;
 
-  /**
-   * Error checking: find strings in string list 1 that are not found in string list 2
-   * @param strList1
-   * @param strList2
-   * @return return strings from strList1 not found in strList2
-   **/
-  std::string compareVectorsOfStrings(const std::vector<std::string> & strList1,
-                                      const std::vector<std::string> & strList2) const;
+  /// max number of numbered value parameters (neml_variable_value0, neml_variable_value1, ...)
+  static const std::size_t _nvars_max = 8;
 
-  /**
-   * Parse the xml file into a string
-   * @param fname xml file name to be read in
-   * @return string containing file contents
-   **/
-  std::string loadFileIntoString(const FileName & fname) const;
+  /// NEML XML Input
+  std::string _xml;
 
-  /**
-   * Get a list of variable names specified in the xml file
-   * @param xmlStringForNeml string containing xml file
-   * @return list of the variable names found in the xml file.
-   **/
-  std::vector<std::string> listOfVariablesInXml(const std::string & xmlStringForNeml) const;
+  /// Find strings in set 1 that are not found in set 2
+  std::vector<std::string> setDifference(const std::set<std::string> & set1,
+                                         const std::set<std::string> & set2) const;
 
-  /**
-   * Check that all variable names in xml file are also in the input file and vice versa
-   * @param nemlNames variables names from input file
-   * @param xmlStringForNeml string containing the xml file
-   **/
-  void errorCheckVariableNamesFromInputFile(const std::vector<std::string> & nemlNames,
-                                            const std::string & xmlStringForNeml) const;
+  /// Build a set of variable names specified in the xml file
+  void findXMLVariables();
 
-  /**
-   * Substitute values in for the variables names found in the string version of the xml file
-   * @param xmlStringForNeml string containing the xml file, returned version of this string has
-   *values substituted in for variables names
-   **/
+  /// Check that all variable names in xml file are also in the input file and vice versa
+  void errorCheckXMLVariables() const;
 
-  void replaceXmlVariables(std::string & xmlStringForNeml) const;
+  /// Substitute values in for the variables names found in the string version of the xml file
```

Comment:
Have we dropped the standard of documenting the methods with the `/**` doxygen comments?  I personally don't like it, but I thought that was the MOOSE coding standard.

Reply:
`///` are single line doxygen comments!

Reply:
I know, but I swear that the style guidelines used to require that we use multiline doxygen comments with `@param`, etc.

Reply:
Now it just says "Try to document as much as possible" (but gives a multiline doxygen example)!